src/App.jsx

```diff
@@ -13,34 +16,90 @@
   'Jam',
   'Garlic',
 ];
+const SORT_FIELD_ALPH = 'alph';
+const SORT_FIELD_LENGTH = 'length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function setSortGoods(goods, sortField) {
+  const sortGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    sortGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPH:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return sortGoods;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  let visibleGoods = setSortGoods(goodsFromServer, sortField);
```

Comment:
The variable `visibleGoods` is correctly initialized with the sorted goods. However, since `setSortGoods` is a pure function, consider using `useMemo` to memoize the sorted goods to avoid unnecessary recalculations on each render.